highlighters/main/main-highlighter.zsh

```diff
@@ -822,15 +833,15 @@ _zsh_highlight_main_highlighter_highlight_list()
       esac
      fi
      if [[ -n ${(M)ZSH_HIGHLIGHT_TOKENS_CONTROL_FLOW:#"$arg"} ]]; then
-      next_word=':start:'
+      next_word=':start::start_of_pipeline:'
      fi
    else # $arg is a non-command word
       case $arg in
         $'\x29') # subshell or end of array assignment
                  if $in_array_assignment; then
                    style=assign
                    in_array_assignment=false
-                   next_word+=':start:'
+                   next_word+=':start::start_of_pipeline:'
```

Comment:
`foo=(bar) ! false` is a parse error here, so I think this is incorrect.

Reply:
Fixed.